ray-operator/config/samples/tpu/tpu_list_devices.py

```diff
@@ -1,19 +1,12 @@
 import ray
+import jax
 
-ray.init(
-    runtime_env={
-        "pip": [
-            "jax[tpu]==0.4.33",
-            "-f https://storage.googleapis.com/jax-releases/libtpu_releases.html",
-        ]
-    }
-)
+ray.init()
 
 @ray.remote(resources={"TPU": 4})
 def tpu_cores():
-    import jax
     return "TPU cores:" + str(jax.device_count())
 
-num_workers = int(os.environ['TPU_HOSTS']) # Set in env of RayJob or RayCluster.
+num_workers = int(ray.available_resources()["TPU"])
```

Comment:
Do you need to "// 4"?

Reply:
Curious what output were you seeing without it? I would expect some sort of error or the job never finishing because there's not enough TPU resources for the remote tasks?

Reply:
This was the output I saw with the above code (except I added a line above line 10 to print num_workers):
```
2024-10-15 20:59:29,675 INFO cli.py:307 -- Tailing logs until the job exits (disable with --no-wait):
2024-10-15 20:59:29,492 INFO job_manager.py:528 -- Runtime env is setting up.
2024-10-15 20:59:39,636 INFO worker.py:1461 -- Using address 10.84.0.18:6379 set in the environment variable RAY_ADDRESS
2024-10-15 20:59:39,636 INFO worker.py:1601 -- Connecting to existing Ray cluster at address: 10.84.0.18:6379...
2024-10-15 20:59:39,642 INFO worker.py:1777 -- Connected to Ray cluster. View the dashboard at 10.84.0.18:8265 
16
['TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16']
2024-10-15 21:00:03,789 SUCC cli.py:63 -- -----------------------------
2024-10-15 21:00:03,789 SUCC cli.py:64 -- Job 'tpu-job-g7tch' succeeded
2024-10-15 21:00:03,789 SUCC cli.py:65 -- -----------------------------
```

Reply:
thanks!

Reply:
It was just printing it too many times, `adding // 4` results in:
```
2024-10-15 21:57:18,284 INFO worker.py:1777 -- Connected to Ray cluster. View the dashboard at 10.84.0.23:8265 
4
['TPU cores:16', 'TPU cores:16', 'TPU cores:16', 'TPU cores:16']
```
The TPU resources were used/freed pretty quickly so it just looped over the devices 4 extra times.

Reply:
Was that with v6e-256 or v6e-16?

Reply:
v6e-16, so 4 workers. I fixed it in c4a233c.